### ISSUE TITLE
! LaTeX Error: Too deeply nested.

### ISSUE THREAD

**Issue by dumblob** [jgm/pandoc#2922]
This is already a long-living issue (there has been discussions [[1]](https://groups.google.com/forum/#!msg/pandoc-discuss/RGSJcRxH1Po/mTQYuIft8pEJ ) [[2]](https://groups.google.com/forum/#!topic/pandoc-discuss/c8nVSifi3NI ) quite a while ago about this), but because it's still not solved (the hack mentioned in [1] obviously doesn't work when not working with markdown or when just converting random documents to PDF) and because I could not find any existing issue here on GitHub, I'm creating one.

I fully understand and even more agree with the rule to make *pandoc* PDF export work even with the *texlive-latex-base* debian package and therefore I would recommend trying to use the [solution outlined on SO](http://stackoverflow.com/a/1937667 ).

**Comment by jgm** [comment]
You mean the solution of adding a bunch of low-level latex definitions to the default latex template?  I'd rather not do that; it adds a lot of length and complexity to the template.

Perhaps we could include a conditional in the template to use enumitem if a certain variable (`enumitem`) is set, and the LaTeX writer could be told to set that variable if the document contains any lists nested more than six levels deep?  The writer would have to keep track of nesting levels as it rendered lists, but this could be done.

This solution would mean that in most cases the enumitem package wouldn't be needed.

**Comment by dumblob** [comment]
>You mean the solution of adding a bunch of low-level latex definitions to the default latex template?

Right, that's what I meant.

>Perhaps we could include a conditional in the template to use enumitem if a certain variable (enumitem) is set, and the LaTeX writer could be told to set that variable if the document contains any lists nested more than six levels deep?

How about making it completely unattended using [IfFileExists](https://tex.stackexchange.com/questions/15306/conditional-if-a-package-is-available ) or [@ifpackageloaded](https://tex.stackexchange.com/questions/16199/test-if-a-package-or-package-option-is-loaded )?

**Comment by jgm** [comment]
+++ dumblob [May 17 16 10:53 ]:
>     Perhaps we could include a conditional in the template to use
>     enumitem if a certain variable (enumitem) is set, and the LaTeX
>     writer could be told to set that variable if the document contains
>     any lists nested more than six levels deep?
>
>   How about making it completely unattended using [1]IfFileExists or
>   [2]@ifpackageloaded?

If we do that, then people who don't have enumitem installed
will still get the LaTeX error, and won't know how to fix
it.  With the solution I suggested, they'd get a different
LaTeX error, but one that points towards a solution (package
enumitem missing).

**Comment by dumblob** [comment]
>If we do that, then people who don't have enumitem installed will still get the LaTeX error, and won't know how to fix it. With the solution I suggested, they'd get a different LaTeX error, but one that points towards a solution (package enumitem missing).

That depends on whether the `enumitem` variable will be set by default or unset. Probably unset (because deeper nesting than 5 shouldn't be more common than nesting up to 5th level) and then you're right.

**Comment by dumblob** [comment]
Any news on this?

**Comment by jgm** [comment]
No, it's not a high priority for me, but maybe someone else wants to code up the solution mentioned?

**Comment by dumblob** [comment]
Well, the part

>The writer would have to keep track of nesting levels as it rendered lists, but this could be done.

seems to be the most difficult one of the whole solution you proposed [above](https://github.com/jgm/pandoc/issues/2922#issuecomment-219770532 ). Could you help at least with this keeping track of nesting levels?

**Comment by jgm** [comment]
+++ dumblob [Nov 06 17 17:08 ]:
>   Well, the part
>
>     The writer would have to keep track of nesting levels as it rendered
>     lists, but this could be done.
>
>   seems to be the most difficult one of the whole solution you proposed
>   [1]above. Could you help at least with this keeping track of nesting
>   levels?

You'd need to add a counter for the nesting level to
LaTeXState.  Then you'd use 'modify' to increment this
before parsing the contents of a list, and decrement
it after.  And you'd use 'gets' to get the current nesting
level.  I suppose you'd need something else in state that
would be set to True if the nesting level ever gets above 5.

**Comment by nathanshelly** [comment]
As a follow up here, while it sounds like fixing this sustainably may not be possible at this time is there a workaround (however hacky)? I noticed an [old post on Gmane about this topic](http://thread.gmane.org/gmane.text.pandoc/4447) where @jgm you seemed to have a temporary solution involving replacing the enumerate package with enumitem. That post is back from 2012 and the template default.latex no longer appears to use the enumerate package. Is a similar workaround still viable and if so what would be the lines now to remove and add? Thanks in advance!

**Comment by jgm** [comment]
Create a file deeplists.tex with the following contents:
`````
   \usepackage{enumitem}
   \setlistdepth{9}

   \setlist[itemize,1]{label=$\bullet$}
   \setlist[itemize,2]{label=$\bullet$}
   \setlist[itemize,3]{label=$\bullet$}
   \setlist[itemize,4]{label=$\bullet$}
   \setlist[itemize,5]{label=$\bullet$}
   \setlist[itemize,6]{label=$\bullet$}
   \setlist[itemize,7]{label=$\bullet$}
   \setlist[itemize,8]{label=$\bullet$}
   \setlist[itemize,9]{label=$\bullet$}
   \renewlist{itemize}{itemize}{9}

   \setlist[enumerate,1]{label=$\arabic*.$}
   \setlist[enumerate,2]{label=$\alph*.$}
   \setlist[enumerate,3]{label=$\roman*.$}
   \setlist[enumerate,4]{label=$\arabic*.$}
   \setlist[enumerate,5]{label=$\alpha*$}
   \setlist[enumerate,6]{label=$\roman*.$}
   \setlist[enumerate,7]{label=$\arabic*.$}
   \setlist[enumerate,8]{label=$\alph*.$}
   \setlist[enumerate,9]{label=$\roman*.$}
   \renewlist{enumerate}{enumerate}{9}
`````

Then use `-H deeplists.tex` on your pandoc command line. That should be all it takes to get deeply nested lists to work.  You need to make sure the enumitem latex package is available.

**Comment by nathanshelly** [comment]
Exactly what I was looking for, thank you so much for the quick reply!

**Comment by craigbass76** [comment]
Is this still valid? I'm continuing to get Too deeply nested errors.

**Comment by undergroundwires** [comment]
Trying to solve this as well. However I cannot install `enumitem` using tlgmr 😟 see jbezos/enumitem#22.

**Comment by undergroundwires** [comment]
Trying to solve this as well. However I cannot install `enumitem` using tlgmr 😟 see: jbezos/enumitem#22.
I wonder if we can solve this without any dependency 🤔

**Comment by ghuser** [comment]
I get `enumitem.sty` not found

**Comment by zedzedzed** [comment]
> I get `enumitem.sty` not found. I run through [docker](https://hub.docker.com/r/pandoc/latex)

I'm also running via Docker and I ran into this problem too. I managed to get around it by downloading `enumitem.sty` from https://ctan.org/tex-archive/macros/latex/contrib/enumitem/ and placing it somewhere within your path (I used my project root).
